devel/xapian-core/Portfile

```diff
@@ -208,6 +208,12 @@ foreach v {2.7 3.6 3.7 3.8 3.9 3.10 3.11} {
     "
 }
 
+if {${subport} eq "xapian-bindings-python36"} {
+    if {${os.platform} eq "darwin" && ${os.major} >= 20} {
+        supported_archs x86_64
```

Comment:
It doesn't support ppc, ppc64, or i386 anymore either?

I dislike blocks like this. You're suggesting that every port that depends on python36, or any other port marked as not supporting arm64, should have a block such as this. I think that would be crummy.

Reply:
@ryandesign I think dependents of Python 3.6 should not be built on arm64 (since Python 3.6 itself is broken), but they are, and CI fail. I have no idea why Macports behaves this way, it looks like a bug.

I have asked about that below, but got no answer.

This chunk should not be needed, but without it CI fail.

P. S. On `${os.major} >= 20` there are no ppc/ppc64 and i386.

Reply:
@ryandesign Maybe you can help with doing this right way?

Only two commits here are essential (adding Ruby 3.3 and a fix for `AI_NUMERICSERV`). The third one was added just because supported_archs does not work for dependents of Python for w/e reason.

Reply:
> I have no idea why Macports behaves this way, it looks like a bug.

https://trac.macports.org/ticket/63092

The short answer is that it would require changing dependency calculation from a relatively simple graph traversal to a backtracking search.

It's probably possible for mpbb to skip ports that will fail for this reason, but someone would have to write the code to implement that. TBH it's a lot easier to just move ports off of EOL python versions.

Reply:
@jmroot @ryandesign Okay, got it. How should we proceed with this one? I am not touching Python ports normally, and even here would prefer not to, but unfortunately this is glued into one portfile.
Obviously, if nobody objects, just removing python36 subport here is an easy solution.

Reply:
Python 3.6 is long since out of support, as is python 3.7. They should both be removed.